lib/OpenQA/Schema.pm

```diff
@@ -100,19 +87,16 @@ sub deploy {
 }
 
 # Class attribute used for testing with OpenQA::Test::Database
-sub search_path_for_tests {
-    my $class = shift;
+sub search_path_for_tests ($class, $new_search_path = undef) {
     state $search_path;
-    $search_path = shift if @_;
+    $search_path = $new_search_path if defined $new_search_path;
```

Comment:
Just an addition to my previous comment:
With this we loose the functionality to set the path to undef.
Previously this was possifle by checking the length of `@_`.
It would still be possible right now and for the next perl versions, but at some point `@_` will be removed for signature subs (there will be an alternative for checking the length of the argument list though).
Context: https://www.nntp.perl.org/group/perl.perl5.porters/2021/12/msg262109.html

So if we need to set it to undef we should check `@_ > 1` here for now.

Reply:
so … change it how exactly? Sorry, can you provide a code suggestion for stupid okurz's? :)

Reply:
There might be better solutions 🤔 , but one possibility would be:
```perl
sub search_path_for_tests ($class, @new_search_path) {
    state $search_path;
    $search_path = $new_search_path[0] if @new_search_path;
    return $search_path;
}
```

Reply:
ok, did that now

Reply:
That's not the code you used.

Reply:
```
sub search_path_for_tests ($class, $new_search_path = undef) {
    state $search_path;
    $search_path = $new_search_path[0] if @new_search_path;
    return $search_path;
}
```

Reply:
For now I left out the changes in OpenQA::Schema from this PR. Will reserve this for a later change. I noted this conversation to be resolved in the according change first.